spaceship/lib/spaceship/client.rb

```diff
@@ -403,12 +421,24 @@ def send_shared_login_request(user, password)
           # which is common, as the session automatically invalidates after x hours (we don't know x)
           # In this case we don't actually care about the exact exception, and why it was failing
           # because either way, we'll have to do a fresh login, where we do the actual error handling
+          puts("Available session is not valid any more. Continuing with normal login.")
         end
       end
-
-      # If this is a CI, the user can pass the session via environment variable
-      # This is used for 2FA related sessions
-      load_session_from_env
+      #
+      # The user can pass the session via environment variable (Mainly used in CI environments)
+      if load_session_from_env
+        # see above
+        begin
+          # see above
+          return true if fetch_olympus_session
+        rescue
+          puts("Session loaded from environment variable is not valid. Continuing with normal login.")
+          # see above
+        end
+      end
+      #
+      # After this point, we sure have no valid session any more and have to create a new one
```

Comment:
```suggestion
      # After this point, we no longer have a valid session, and have to create a new one
```

Reply:
```suggestion
      # After this point, we certainly don't have a valid session and have to create a new one
```
What do you think about this one @getaaron?